Index: qt-advanced-docking-system/src/CMakeLists.txt
===================================================================
--- qt-advanced-docking-system.orig/src/CMakeLists.txt
+++ qt-advanced-docking-system/src/CMakeLists.txt
@@ -91,12 +91,6 @@ install(FILES ${ads_HEADERS}
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/qtadvanceddocking
     COMPONENT headers
 )
-install(FILES
-    "${CMAKE_CURRENT_SOURCE_DIR}/../LICENSE"
-    "${CMAKE_CURRENT_SOURCE_DIR}/../gnu-lgpl-v2.1.md"
-    DESTINATION license/ads
-    COMPONENT license
-)
 install(TARGETS qtadvanceddocking
     EXPORT adsTargets
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
